c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -888,76 +885,90 @@ impl TypedAstContext {
-                                && !rhs_resolved_ty.kind.is_pointer();
-
-                            if op.all_types_same() && neither_ptr {
-                                if CTypeKind::PULLBACK_KINDS.contains(&lhs_resolved_ty.kind) {
-                                    Some(lhs_type_id)
-                                } else {
-                                    Some(rhs_type_id)
-                                }
-                            } else if op == BinOp::ShiftLeft || op == BinOp::ShiftRight {
-                                Some(lhs_type_id)
                             } else {
-                                return;
+                                Some(rhs_type_id)
                             }
+                        } else if op == BinOp::ShiftLeft || op == BinOp::ShiftRight {
+                            Some(lhs_type_id)
+                        } else {
+                            return;
                         }
-                        CExprKind::Unary(_ty, op, e, _idk) => op.expected_result_type(
-                            self.ast_context,
-                            self.ast_context.c_exprs[&e].kind.get_qual_type().unwrap(),
-                        ),
-                        CExprKind::Paren(_ty, e) => {
-                            self.ast_context.c_exprs[&e].kind.get_qual_type()
-                        }
-                        _ => return,
-                    };
-                    if let (Some(ty), Some(new_ty)) = (
-                        self.ast_context
-                            .c_exprs
-                            .get_mut(&e)
-                            .and_then(|e| e.kind.get_qual_type_mut()),
-                        new_ty,
-                    ) {
-                        *ty = new_ty;
-                    };
-                }
+                    }
+                    CExprKind::Unary(_ty, op, e, _idk) => op.expected_result_type(
+                        self.ast_context,
+                        self.ast_context.c_exprs[&e].kind.get_qual_type().unwrap(),
+                    ),
+                    CExprKind::Paren(_ty, e) => self.ast_context.c_exprs[&e].kind.get_qual_type(),
+                    CExprKind::UnaryType(_, op, _, _) => {
+                        // All of these `UnTypeOp`s should return `size_t`.
+                        let kind = match op {
+                            UnTypeOp::SizeOf => CTypeKind::Size,
+                            UnTypeOp::AlignOf => CTypeKind::Size,
+                            UnTypeOp::PreferredAlignOf => CTypeKind::Size,
+                        };
+                        let ty = self
+                            .ast_context
+                            .type_for_kind(&kind)
+                            .expect("CTypeKind::Size should be size_t");
+                        Some(CQualTypeId::new(ty))
+                    }
+                    _ => return,
+                };
+                let ty = self
```

Comment:
Isn't this the same expression, with a refactoring? Do you need to change this?

Reply:
It was just harder to read since the end of the method chain and if block were far apart.